### ISSUE TITLE
I tap any App in Main, Select Clone, App Doesn't Appear in Shelter area.

### ISSUE THREAD

**Issue by mawil1013** [PeterCxy/Shelter#128]
Phone is stock OS OnePlus 7T Pro.
I go to Main to select any App to Clone, message to Clone pops up and I select Clone to Shelter, but none appear in Shelter list.  ( FYI: I terminated Shelter to install Island and Island works.)

**Comment by PeterCxy** [comment]
See #122. Shelter is currently missing the progress bar of cloning apps. You have to wait a while (without closing Shelter) for the cloning process to actually complete. I will resolve this, but probably not now, because issues mentioned in pinned posts.

**Comment by Allba** [comment]
> Phone is stock OS OnePlus 7T Pro.
> I go to Main to select any App to Clone, message to Clone pops up and I select Clone to Shelter, but none appear in Shelter list. ( FYI: I terminated Shelter to install Island and Island works.)

I have the same problem with the exact same phone, waiting for the app to clone doesn't seem to work on my phone.

**Comment by PeterCxy** [comment]
@Allba  Please see #122 and #129. Currently the installation progress bar is simply missing. Please do not close Shelter after clicking clone -- the status bar and navigation bar will turn black for a while. When the black is gone, the application will be cloned.

**Comment by TechInterMezzo** [comment]
This must be another bug. I have the same problem sometimes. I can wait a whole minute and nothing happens. I have to restart the app then for cloning to work again.

**Comment by projectdallas** [comment]
I think it's another bug too.  Cloning may or may not work depending on the app.

1. I select an app
2. Tap on _Clone to Shelter (Work Profile)_
3. Sometimes it may prompt "Do you want to install this application?" and "Blocked by Play Protect: this app may be harmful". I tap Install.
4. A black bar slides from top to bottom of the screen.

I go to the Shelter tab. App isn't on the list.

I'm using LineageOS (Android 10) with root.

Is there a way to debug?

**Comment by PeterCxy** [comment]
@projectdallas the "black bar" is exactly the symptom of the missing progress bar. It's exactly what I was talking about.

**Comment by projectdallas** [comment]
> 
> 
> @projectdallas the "black bar" is exactly the symptom of the missing progress bar. It's exactly what I was talking about.

My experience is odd. I tapped _Clone to Shelter_ and kept the Shelter app running (with screen on) for 10 minutes. I checked the Shelter tab. Nothing.

I can clone all stock apps (clock, calendar, FM radio etc) with **no** problems! But every downloaded app from F-Droid or Google Play, cloning is unreliable.

I did a full device wipe via TWRP, flashed ROM, Gapps and Magisk. Downloaded F-Droid, and then DuckDuckGo and Shelter apps. Attempted to clone DuckDuckGo. Nothing :\

When installing Shelter, one thing was odd. It will say _Please wait while we prepare Shelter for you ..._ with a spinner. And then a notification says _Click here to finish setting up Shelter_. I waited for 10 minutes but it kept spinning until I tapped on the notification. Is this normal?

So something could be wrong with the installation, or the ROM, or the app...

**Comment by PeterCxy** [comment]
@projectdallas 

> I can clone all stock apps (clock, calendar, FM radio etc) with no problems! But every downloaded app from F-Droid or Google Play, cloning is unreliable.

Yes that's the symptom of the aforementioned problem. For now the only workaround is to wait or retry several times. This will be resolved once I get back to this project.

> When installing Shelter, one thing was odd. It will say Please wait while we prepare Shelter for you ... with a spinner. And then a notification says Click here to finish setting up Shelter. I waited for 10 minutes but it kept spinning until I tapped on the notification. Is this normal?

You should have clicked on the notification the minute it came out.

**Comment by PeterCxy** [comment]
Also, to everyone else that may see this problem, another workaround for now is to install the apk via `adb install`. You can install a third-party store or something this way into Shelter and use that to install other applications.

**Comment by Allba** [comment]
> 
> 
> Also, to everyone else that may see this problem, another workaround for now is to install the apk via `adb install`. You can install a third-party store or something this way into Shelter and use that to install other applications.

Is there a specific command to install it on Shelter via adb?

**Comment by PeterCxy** [comment]
@Allba `adb install` installs to both the main user and the work profile by default. You can then uninstall it manually in main user.

**Comment by PeterCxy** [comment]
Fixed in 3395f55 and 7774395
